packages/block-editor/src/components/link-control/index.js

```diff
@@ -266,29 +266,40 @@ function LinkControl( {
 			{}
 		);
 
-		onChange( {
+		setInternalControlValue( {
 			...internalControlValue,
 			...nonSettingsChanges,
 			// As title is not a setting, it must be manually applied
 			// in such a way as to preserve the users changes over
 			// any "title" value provided by the "suggestion".
 			title: internalControlValue?.title || updatedValue?.title,
+			showEdit: true,
 		} );
 
 		stopEditing();
 	};
 
 	const handleSubmit = () => {
 		if ( valueHasChanges ) {
-			// Submit the original value with new stored values applied
-			// on top. URL is a special case as it may also be a prop.
-			onChange( {
+			setInternalControlValue( {
 				...value,
 				...internalControlValue,
+				title: internalControlValue?.title,
 				url: currentUrlInputValue,
+				showEdit: true,
 			} );
 		}
-		stopEditing();
+	};
+
+	const handleSave = () => {
+		// Submit the original value with new stored values applied
+		// on top. URL is a special case as it may also be a prop.
+		onChange( {
+			...value,
+			...internalControlValue,
+			url: currentUrlInputValue,
+			showEdit: false,
```

Comment:
The value we return from LinkControl has a standardised shape and I don't think we should expose this type of internal rendering state on the value object.

Why? Because they are different concerns.

One is data.

One is rendering state.

If one day we decide on a different UX then this `showEdit` will become a redundant property to support forever.